src/IO/S3/URI.cpp

```diff
@@ -101,6 +102,16 @@ URI::URI(const std::string & uri_)
     }
     else
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Bucket or key name are invalid in S3 URI.");
+
+    if(key.empty()){
+        key = "default/";
```

Comment:
Do you actually need to have a `default/` key in this case? Can't you just use root of the bucket?

Reply:
I think this way, the data can be uploaded more neatly, and it is also convenient to search in S3. Of course, it can be changed here